core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/relational/SatisfiesExpression.kt

```diff
@@ -0,0 +1,87 @@
+package ch.ergon.dope.resolvable.expression.unaliased.type.relational
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.expression.unaliased.type.relational.SatisfiesType.ANY
+import ch.ergon.dope.resolvable.expression.unaliased.type.relational.SatisfiesType.EVERY
+import ch.ergon.dope.resolvable.expression.unaliased.type.toArrayType
+import ch.ergon.dope.validtype.ArrayType
+import ch.ergon.dope.validtype.BooleanType
+import ch.ergon.dope.validtype.ValidType
+
+object IteratorManager {
+    var count: Int = 1
+        get() = field++
+
+    fun resetCounter() {
+        count = 1
+    }
+}
+
+const val DEFAULT_ITERATOR_VARIABLE = "iterator"
```

Comment:
Why do we need a default iterator variable?

Reply:
With the Any/Every Satisfies it doesn't really matter, what iterator variable u use, because it will only be used in this one query.